app/src/androidTest/java/com/sdp/movemeet/view/navigation/NavigationTest.java

```diff
@@ -0,0 +1,121 @@
+import android.view.Gravity;
+
+import androidx.annotation.NonNull;
+import androidx.test.core.app.ActivityScenario;
+import androidx.test.espresso.contrib.DrawerActions;
+import androidx.test.espresso.contrib.NavigationViewActions;
+import androidx.test.espresso.intent.Intents;
+import androidx.test.espresso.matcher.ViewMatchers;
+import androidx.test.ext.junit.rules.ActivityScenarioRule;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.rule.ActivityTestRule;
+
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.firebase.auth.AuthResult;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.sdp.movemeet.R;
+import com.sdp.movemeet.backend.providers.AuthenticationInstanceProvider;
+import com.sdp.movemeet.backend.providers.BackendInstanceProvider;
+import com.sdp.movemeet.view.activity.UploadActivityActivity;
+import com.sdp.movemeet.view.chat.ChatActivity;
+import com.sdp.movemeet.view.home.LoginActivity;
+import com.sdp.movemeet.view.main.MainActivity;
+import com.sdp.movemeet.view.profile.ProfileActivity;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.concurrent.CountDownLatch;
+
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.action.ViewActions.closeSoftKeyboard;
+import static androidx.test.espresso.action.ViewActions.replaceText;
+import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.contrib.DrawerMatchers.isClosed;
+import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Matchers.intThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(AndroidJUnit4.class)
+public class NavigationTest {
+
+    private FirebaseAuth fAuth;
+    private final String TAG = "Navigation test";
+
+    @Rule
+    public ActivityScenarioRule<LoginActivity> testRule = new ActivityScenarioRule<>(LoginActivity.class);
+
+    @Before
+    public void signIn() throws InterruptedException {
+        MainActivity.enableNav = false;
+        fAuth = FirebaseAuth.getInstance();
+        onView(withId(R.id.edit_text_email)).perform(replaceText("antho2@gmail.com"), closeSoftKeyboard());
+        onView(withId(R.id.edit_text_password)).perform(replaceText("234567"), closeSoftKeyboard());
+        onView(withId(R.id.button_login)).perform(click());
+        sleep();
+    }
+
+    @Test
+    public void NavigationToHomeTest() {
```

Comment:
Shouldn't these tests assert that the correct intent is fired every time?

Reply:
Yes indeed, I will add it ! I hadn't this asserts first, just to be sure it was working (we had quite a few problems with those tests).